ops/pebble.py

```diff
@@ -1011,27 +1013,30 @@ def close(self):
 class _WebsocketReader(io.BufferedIOBase):
     """A readable file-like object whose reads come from a websocket."""
 
-    def __init__(self, ws, encoding):
+    def __init__(self, ws):
         self.ws = ws
-        self.encoding = encoding
-        self.remaining = '' if encoding is not None else b''
+        self.remaining = b''
+        self.eof = False
 
     def readable(self):
         """Denote this file-like object as readable."""
         return True
 
     def read(self, n=-1):
         """Read up to n bytes from the websocket (or one message if n<0)."""
+        if self.eof:
+            # Calling read() multiple times after EOF should still return EOF
+            return b''
+
         if not self.remaining:
             recv = self.ws.recv()
 
             if isinstance(recv, str):
                 _ = json.loads(recv)  # raise ValueError on invalid JSON
```

Comment:
So we check it's JSON but don't check the value. I can't remember what we chose on the Pebble side, but I thought it was more complex

Reply:
Updated to check the command and ignore in both cases, but log a warning (same as the Go client).